engine/src/main/java/org/terasology/world/block/tiles/WorldAtlasImpl.java

```diff
@@ -250,9 +269,18 @@ private void createTextureAtlas(final Texture texture) {
     }
 
     private ByteBuffer[] createAtlasMipmaps(int numMipMaps, Color initialColor, List<BlockTile> tileImages, String screenshotName) {
+        return createAtlasMipmaps(numMipMaps, initialColor, tileImages, screenshotName, Lists.newArrayList());
+    }
+
+    private ByteBuffer[] createAtlasMipmaps(int numMipMaps, Color initialColor, List<BlockTile> tileImages, String screenshotName, List<BlockTile> alphaMaskTiles) {
         ByteBuffer[] data = new ByteBuffer[numMipMaps];
         for (int i = 0; i < numMipMaps; ++i) {
             BufferedImage image = generateAtlas(i, tileImages, initialColor);
+            if (alphaMaskTiles.size() > 0) {
+                BufferedImage alphaMask = generateAtlas(i, alphaMaskTiles, Color.BLACK);
+                storeGreyscaleMapIntoAlpha(image, alphaMask);
```

Comment:
I feel there are two reasons to keep the method call this generic:

1. it already deals with more than one specific texture type-pair.
2. you can _envision_ this method being used for more than one specific texture types-pair. And I'm not talking about hypotheticals, I'm talking about something you have in mind but can't quite implement yet or simply doesn't make sense adding to this PR.

If either one of the above statements is true then I'd be ok with the current version.

However, if the only texture type-pair going through this code is normalsMap/specularMap AND you can't think of any other type-pair soon going through this code... I'd then feel strongly about tweaking those variable names. =)